lib/deimos/utils/schema_class.rb

```diff
@@ -10,10 +10,23 @@ class << self
         # @return [Array<String>]
         def modules_for(namespace)
           modules = ['Schemas']
-          namespace_folder = namespace.split('.').last
-          if Deimos.config.schema.generate_namespace_folders && namespace_folder
-            modules.push(namespace_folder.underscore.classify)
+
+          if Deimos.config.schema.generate_namespace_folders
+            namespace_map = Deimos.config.schema.schema_namespace_map.sort_by{|k,v| -k.length}.to_h
+            namespace_override = namespace_map.keys.detect{ |k| namespace.include?(k) }
+
+            module_namespace = if namespace_override.present?
```

Comment:
This block is setting a return value and also doing side effects... I'd prefer this in the format of

```
module_namespace = namespace
if namespace_override.present?
  module_namespace = namespace.gsub(/#{namespace_override}\.?/, '')
  modules = [namespace_map[namespace_override]]
end
```

Reply:
sure, will update